### PR TITLE
feat: Add lu_unpack function to torch frontend

### LINKED ISSUE #26225: lu_unpack

### PR DESCRIPTION
**Pull request by p1utoze**
# PR Description

The lu_unpack function is limited to scipy, torch and paddle. As there are no backend functions implemented yet in Experimental API, I have written a custom and generic unpack for the 2D LU matrix along with the conversion of pivot indices to permutation matrix. 
The multi-dimensional lu_unpack is a bit more complex and cluttered in the frontend function. It's better if it is implemented in the Experimental API.

## Related Issue

Resolve #26225

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

File: ivy/functional/frontends/torch/blas_and_lapack_ops.py [hunk 1]
```diff
@@ -130,6 +130,46 @@ def logdet(input):
...
+    m, n = lu_data.shape[-2:]
+    k = min(m, n)
+    if unpack_pivots:
+        permute = ivy.arange(m)
+        for i in range(lu_pivots.shape[-1]):
+            if i != lu_pivots[i] - 1:
+                permute[i], permute[lu_pivots[i] - 1] = (
+                    permute[lu_pivots[i] - 1],
+                    permute[i],
+                )
```
Comment (KareemMAX) [inline ivy/functional/frontends/torch/blas_and_lapack_ops.py:154]: Can you find a vectorized version of this code? I know it's `O(K)` but I think if there is a vectorized version would be better.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_blas_and_lapack_ops.py [hunk 2]
```diff
@@ -225,6 +225,29 @@ def _get_dtype_input_and_vectors(draw, with_input=False, same_size=False):
     return dtype, vec1, vec2
 
 
+@st.composite
+def _get_dtype_lu_and_pivots(draw):
+    dimsize1 = draw(st.integers(min_value=2, max_value=5))
+    dimsize2 = draw(st.integers(min_value=2, max_value=5))
+    # dtype = draw(helpers.get_dtypes("valid"))
+    dtype = ["float32", "float64"]
```
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_blas_and_lapack_ops.py:233]: I guess you should use this
```suggestion
    dtype = draw(helpers.get_dtypes("valid"))
```

File: ivy_tests/test_ivy/test_frontends/test_torch/test_blas_and_lapack_ops.py [hunk 3]
```diff
@@ -225,6 +225,29 @@ def _get_dtype_input_and_vectors(draw, with_input=False, same_size=False):
...
+    # dtype = [
+    #     draw(st.sampled_from(tuple(set(dtype).difference({"bfloat16", "float16"}))))
+    # ]
```
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_blas_and_lapack_ops.py:236]:
```suggestion
```

File: ivy_tests/test_ivy/test_frontends/test_torch/test_blas_and_lapack_ops.py [hunk 4]
```diff
@@ -225,6 +225,29 @@ def _get_dtype_input_and_vectors(draw, with_input=False, same_size=False):
...
+
+    mat = draw(
+        helpers.array_values(
+            dtype=dtype[0], shape=(dimsize1, dimsize2), min_value=1, max_value=10
+        )
+    )
```
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_blas_and_lapack_ops.py:242]: Why don't you use [`dtype_and_values`](https://unify.ai/docs/ivy/docs/helpers/ivy_tests.test_ivy.helpers.hypothesis_helpers/ivy_tests.test_ivy.helpers.hypothesis_helpers.array_helpers.html#ivy_tests.test_ivy.helpers.hypothesis_helpers.array_helpers.dtype_and_values)?

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hey @p1utoze,
Good job 🚀, but I noticed you have deleted some irrelevant files. Can you revert that change so we can proceed with the review?

I've pointed out some issues I would like you to resolve while I was skimming the PR. Please request a review once you revert those changes

File: ivy/functional/frontends/torch/blas_and_lapack_ops.py (hunk 1 above)
Comment (p1utoze) [reply ivy/functional/frontends/torch/blas_and_lapack_ops.py:155]: Hey!  I tried vectorizing the above loop using this:
```
permute, indices = ivy.arange(m), ivy.arange(lu_pivots.shape[-1])
mask = ivy.where(pivots != lu_pivots - 1, True, False)
permute[indices], permute[lu_pivots[mask] - 1] = permute[lu_pivots[mask] - 1], permute[indices[mask]]
```
However, there are some edge cases where the pivot indices are repeated and in that case, permuting sequentially gives exact permutation matrix ordering.

**Comment by p1utoze** [comment]
> Hey @p1utoze, Good job 🚀, but I noticed you have deleted some irrelevant files. Can you revert that change so we can proceed with the review?
> 
> I've pointed out some issues I would like you to resolve while I was skimming the PR. Please request a review once you revert those changes

Hi! There's some issue with my local fork and I'm unable to add the deleted files back.

**Comment by KareemMAX** [comment]
> > Hey @p1utoze, Good job 🚀, but I noticed you have deleted some irrelevant files. Can you revert that change so we can proceed with the review?
> > I've pointed out some issues I would like you to resolve while I was skimming the PR. Please request a review once you revert those changes
> 
> Hi! There's some issue with my local fork and I'm unable to add the deleted files back.

Sadly we won't be able to merge with those files deleted, you may try to revert/undo your commits. Or you might copy over your changes to a new fork and create a PR from it.

If you choose the latter please close this PR.

**Comment by p1utoze** [comment]
> 
> Sadly we won't be able to merge with those files deleted, you may try to revert/undo your commits. Or you might copy over your changes to a new fork and create a PR from it.
> 
> If you choose the latter please close this PR.
Yeah, I thought to close this PR. Mainly it has issues in my fork and also the commits aren't clean.

**Pull request closed without merging**